### ISSUE TITLE
Refit?

### ISSUE THREAD

**Issue by hansmbakker** [RoguePlanetoid/Spotify-NetStandard#2]
Nice that you're creating this! I was wondering whether you considered https://github.com/reactiveui/refit to simplify creating your REST client?

**Comment by RoguePlanetoid** [comment]
Thanks - I'd based this client on the Groove Music SDK but anything I can do with it later might be worth considering as a later improvement using something like this!

**Comment by RoguePlanetoid** [comment]
The REST Library has been rewritten to improve the workflow, hopefully this makes it much easier to understand

**Comment by hansmbakker** [comment]
I meant that it might make it much easier for you to maintain this library and to add new functionality if you do not hand-write the REST Client code yourself. Also it might make your code much cleaner.

Refit just asks you to write a .Net interface and specify the REST method, parameters and url route to disclose an endpoint.

**Comment by hansmbakker** [comment]
More specifically, it would make a part of the contents in your [Spotify.NetStandard/Spotify.NetStandard/Client/Internal](https://github.com/RoguePlanetoid/Spotify-NetStandard/tree/master/Spotify.NetStandard/Spotify.NetStandard/Client/Internal) folder redundant.

**Comment by hansmbakker** [comment]
Besides the point about Refit, I was wondering how you think about the circular dependency between SpotifyApi and SpotifyClient. Personally I find it very confusing and it could improve your code if there is one class that makes the REST calls and one class that might add some logic to the REST calls.

**Comment by RoguePlanetoid** [comment]
Good point about that circular dependancy - I'd intended that ISpotifyApi Interface to be a wrapper for the ISpotifyClient one but agree it is confusing and could refactor this to remove that

**Comment by hansmbakker** [comment]
Nice :) I was wondering whether you've had the chance to look at Refit?

My recommended approach would be:
- let `ISpotifyApi` handle all REST calls using Refit attribute decorations
- remove `SimpleServiceClient` - all this code can be generated using Refit so that you do not need to maintain it
- create `ISpotifyClient` / `SpotifyClient` for the extra logic on top of the REST calls (e.g. the modification of parameters, etc). In the implementation, use `ISpotifyApi` to make the API calls.
- use the [authentication pattern from Refit](https://github.com/reactiveui/refit#authorization-dynamic-headers-redux) to handle the authentication part
